mllib/src/test/scala/org/apache/spark/ml/feature/QuantileDiscretizerSuite.scala

```diff
@@ -146,4 +146,166 @@ class QuantileDiscretizerSuite
     val model = discretizer.fit(df)
     assert(model.hasParent)
   }
+
```

Comment:
We should add 2 tests:

1. test setting `numBuckets` is the same as setting `numBucketsArray` explicitly with identical values
2. test that QD over multiple columns produces the same results as 2x QDs over the same columns (as we did for Bucketizer)